### ISSUE TITLE
All images need to have alt text or be marked as decorative.

### ISSUE THREAD

**Issue by bedmison** [OpenDSA/OpenDSA#617]
This applies to every image in the system. The alt text should be meaningful but relatively short. 

Ensure <img> elements have alternate text or a role of none or presentation

Fix:

To solve this problem, you need to fix at least (1) of the following:
- Element does not have an alt attribute
- aria-label attribute does not exist or is empty
-aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
- Element has no title attribute
- Element's default semantics were not overridden with role="none" or role="presentation"

**Comment by cashaffer** [comment]
Looking for examples where this is an actual problem.

In general, I am aware of two sorts of images. The good news is that both are either addressed, or should be easy to address with some minimal programming adjustments.
1) Ones that are generated procedurally as part of book composition. This includes things like logos, checkmarks for completion, etc. So far as I have seen so far, they all get generated with "alt" tags.
2) Actual images loaded in as part of the content. These are loaded with the ".. odsafig" directive. That directive as an "alt" field, that gets compiled as the "alt" text for the image tag. I can't swear that they ALL have this in the source file, but the few that I checked so far all seem to have it. They would be easy to add once we know the few that are missing.

So what I need now are examples of actual problem images.

**Comment by maellis1** [comment]
Thanks Cliff.

Bob - do you have a list to share at this point?  How can I be most helpful?

Cliff - I have a current CS2114 student who is having trouble accessing content so we are trying to address this in many ways.  Thanks!

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by maellis1** [comment]
I shared the following folders with Cliff:


  1.
where we are putting updated CS2114 images: https://drive.google.com/drive/folders/1skXSgmR457IzQL1knoXahlh7uz9z7f7m?usp=drive_link

  2.
where Bob and I are trying to track on the content that needs attention: https://docs.google.com/spreadsheets/d/1BCg5ANLRPfUp_5KD82iMFHGbqVyVUQ_1EXo4gAIIdCY/edit?usp=sharing

________________________________

**Comment by bedmison** [comment]
[OpenDSA_images.txt](https://github.com/user-attachments/files/17117730/OpenDSA_images.txt)

This is a list I got from Master in the repo using: `grep -r --exclude-dir=Books -n "<img" >`

This includes the any libs we may rely on that have some IMG refs without alt text (ie CodeMirror). We prob want to look at updating the versions for these libs. I think this list also includes some legacy code we don't use anymore.

**Comment by cashaffer** [comment]
I see a few places here that need work on the image generators.

This list would be a lot more valuable if we could exclude lines that contain "alt=".

**Comment by s-edwards** [comment]
Add onto the end of the command (or re-grep the output file):    | grep -v alt=

**Comment by bedmison** [comment]
Per Cliff email to Arinjoy, here are the instances in the DeformsTesting when the alt text needs work:

```
./Exercises/DeformsTesting/f2020p10PPRO.html:71:              <img class="figure1" src="./problemfigures/ph2-p10.png" style="width: 500px;" alt="description">
./Exercises/DeformsTesting/S23torsionproblemFeb7.html:77:              <img class="figure1" src="./problemfigures/S23torsionproblemFeb7full.png" alt="Text here">
./Exercises/DeformsTesting/S23ESclevis2compactPPRO.html:70:              <img class="figure1" src="./problemfigures/ph2-p1.png" style="width: 500px;" alt="description">
./Exercises/DeformsTesting/esm2204f2020hw3p1PPRO.html:59:              <img class="figure1" src="./problemfigures/esm2204f2020hw3p1.png" alt="Text here">
./Exercises/DeformsTesting/DeformsDemoCompactProblemPPRO.html:80:              <img class="figure1" src="./problemfigures/demo-problem.png" alt="Text here">
./Exercises/DeformsTesting/S23ESclevis3compactPPRO.html:71:              <img class="figure1" src="./problemfigures/ph2-p1.png" style="width: 500px;" alt="description">
./Exercises/DeformsTesting/f2020p2PPRO.html:60:              <img class="figure1" src="./problemfigures/ph2-p2.png" style="width: 350px;" alt="description">
./Exercises/DeformsTesting/f2020p3PPRO.html:93:              <img class="figure1" src="./problemfigures/ph2-p3.png" style="width: 500px;" alt="description">
./Exercises/DeformsTesting/f2020p8PPRO.html:65:              <img class="figure1" src="./problemfigures/ph2-p8.png" style="width: 500px;" alt="description">
./Exercises/DeformsTesting/f2020p9PPRO.html:76:              <img class="figure1" src="./problemfigures/ph2-p9.png" style="width: 400px;" alt="description">
./Exercises/DeformsTesting/DeformsDemoProblemPPRO.html:74:              <img class="figure1" src="./problemfigures/demo-problem.png" alt="Text here">
./Exercises/DeformsTesting/S23ESclevis1compactPPRO.html:60:              <img class="figure1" src="./problemfigures/ph2-p1.png" style="width: 500px;" alt="description">
./Exercises/DeformsTesting/f2020p8CompactPPRO.html:72:              <img class="figure1" src="./problemfigures/ph2-p8.png" style="width: 500px;" alt="description">
./Exercises/DeformsTesting/f2020p7PPRO.html:135:              <img class="figure1" src="./problemfigures/ph2-p7.png" style="width: 250px;" alt="description">
./Exercises/DeformsTesting/f2020p1PPRO.html:90:              <img class="figure1" src="./problemfigures/ph2-p1.png" style="width: 500px;" alt="description">
```

**Comment by bedmison** [comment]
Here is a list of the JS files that have code where they create an IMG tag in the DOM:
```
./AV/OpenFLAP/exercises/exerController/GrammarExerciseController.js:280
./AV/OpenFLAP/exercises/exerController/GrammarExerciseController.js:292
./AV/OpenFLAP/exercises/exerController/GrammarExerciseController.js:303
./AV/OpenFLAP/exercises/exerController/REExerciseController.js:145
./AV/OpenFLAP/exercises/exerController/REExerciseController.js:157
./AV/OpenFLAP/exercises/exerController/REExerciseController.js:168
./AV/OpenFLAP/exercises/exerController/NFAtoDFAMinimizationController.js:109
./AV/OpenFLAP/exercises/exerController/ExerciseController.js:208
./AV/OpenFLAP/exercises/exerController/ExerciseController.js:213
./AV/OpenFLAP/exercises/exerController/ExerciseController.js:221
./AV/OpenFLAP/exercises/exerController/ExerciseController.js:230
./AV/OpenFLAP/exercises/exerController/ExerciseController.js:241
```
